src/services/storage.js

```diff
@@ -17,135 +17,119 @@
   this.name = name;
-  this.opts = { ...opts };
-  this.connect();
+  this.prefix = `wallet-storage:${this.name}:`;
 }
 
-Storage.prototype.connect = function () {
-  this.needToReconnect = false;
-  this.connection = new PouchDB(this.name, this.opts);
-  console.log(`Connected to the database ${this.name}`);
-};
-
-Storage.prototype.db = function () {
-  if (this.needToReconnect) this.connect();
-  return this.connection;
+Storage.prototype.getKey = function (id: string) {
+  return this.prefix + id;
 };
 
-Storage.prototype.get = function (id: string) {
-  return this.db().get(id).catch(() => ({}));
+Storage.prototype.get = async function (id: string) {
+  const data = await AsyncStorage.getItem(this.getKey(id))
+    .then(JSON.parse)
+    .catch(() => {});
+  return data || {};
 };
 
-Storage.prototype.getConflicts = function (): Promise<String[]> {
-  return this.getAllDocs()
-    .then(({ rows }) => (
-      rows.map(({ doc }) => doc._conflicts).reduce((memo, item) => memo.concat(item), [])
-    ));
-};
+const activeDocs = {};
+Storage.prototype.save = async function (id: string, data: Object, forceRewrite: boolean = false) {
+  const currentValue = await this.get(id);
+  const key = this.getKey(id);
+
+  if (activeDocs[key]) {
+    Sentry.captureMessage('Race condition spotted', {
+      extra: {
+        id,
+        data,
+        forceRewrite,
+      },
+    });
+  }
 
-/**
- * V1 db repairment by dropping the whole db and re-populating it with the old data.
- */
-Storage.prototype.repair = async function () {
-  const docs = await this.getAllDocs().then(({ rows }) => rows.map(({ doc }) => doc));
-  if (!docs.length) return Promise.resolve();
-  await this.db().destroy();
-  this.connect();
-  const promises = docs.map(doc => {
-    const {
-      _id,
-      _rev,
-      _conflicts,
-      ...data
-    } = doc;
-    return this.save(_id, data).catch(() => null);
-  });
-  return Promise.all(promises);
-};
+  activeDocs[key] = true;
 
-const activeDocs = {};
-Storage.prototype.save = function (id: string, data: Object, forceRewrite: boolean = false) {
-  return this.db().get(id)
-    .catch(err => {
-      if (err.status !== 404) {
-        throw err;
-      }
-      return {};
-    })
-    .then(doc => {
-      if (activeDocs[id]) {
-        Sentry.captureMessage('Race condition spotted', {
-          extra: {
-            id,
-            data,
-            forceRewrite,
-          },
-        });
-      }
-
-      activeDocs[id] = true;
-      const options = { force: forceRewrite };
-      const record = forceRewrite
-        ? { _id: id, _rev: doc._rev, ...data }
-        : merge(
-          {},
-          doc,
-          {
-            _id: id,
-            _rev: doc._rev,
-          },
-          data,
-        );
-      return this.db().put(record, options);
-    })
-    .then(doc => {
-      activeDocs[id] = false;
-      return doc;
+  const newValue = forceRewrite ? data : merge({}, currentValue, data);
```

Comment:
It looks like we don't need the `currentValue` unless we have `forceRewrite`.

Maybe having it on a function could make it simpler:

```js
const mergeValue = async (key: string, data: Object): Object => {
  const currentValue = await this.get(id);
  return merge({}, currentValue, data);
};

const newValue = forceRewrite ? data : mergeValue(key, data);
```

Reply:
👍 good suggestion. This should speed up the execution as well